lib/bulbs-elements/util/in-view-monitor.js

```diff
@@ -0,0 +1,65 @@
+const isElementInViewport = (el) => {
+  let rect = el.getBoundingClientRect();
+
+  return rect.bottom > 0 &&
+    rect.right > 0 &&
+    rect.left < (window.innerWidth || document.documentElement.clientWidth) &&
+    rect.top < (window.innerHeight || document.documentElement.clientHeight)
+  ;
+};
+
+let inViewId = 0;
+let initialized = false;
+let animationRequest;
+let all = {};
+
+const init = () => {
+  if (initialized === false) {
+    initialized = true;
+    window.addEventListener('scroll', handleDisplayMutation);
+    window.addEventListener('resize', handleDisplayMutation);
+  }
+};
+
+const maybeEmitEvent = (one) => {
+  let inView = isElementInViewport(one.element);
```

Comment:
Yeah, the `all` collection stashes items that look like:
```js
{
  element: domNode,
  state: 'init' || 'in' || 'out',
}
```

I'm all-in on a more descriptive name, but I'm not sure what would be best here.